packages/slider/src/ux-slider.ts

```diff
@@ -98,36 +98,52 @@ export class UxSlider implements UxComponent {
   }
 
   public updateValue(currentMouseX: number) {
-    const normalizedMouseX = currentMouseX - this.element.offsetLeft;
-    const percentValue = normalizedMouseX / this.element.clientWidth;
+    const rect = this.element.getBoundingClientRect();
+    const normalizedMouseX = currentMouseX - rect.x;
+    const percentValue = normalizedMouseX / rect.width;
     const rawValue = ((this.max - this.min) * percentValue) + this.min;
     const numSteps = Math.round((rawValue - this.min) / this.step);
     const steppedValue = this.min + (this.step * numSteps);
 
     this.value = this.boundValue(steppedValue);
   }
 
-  public onTrackMouseDown() {
+  public onTrackMouseDown(e: MouseEvent | TouchEvent) {
     if (this.disabled) {
       return;
     }
 
     this.isActive = true;
 
     const winEvents = new ElementEvents(window as any);
-    const upAction = (e: MouseEvent) => {
+    const upAction = (e: MouseEvent | TouchEvent | FocusEvent) => {
       if (!this.isActive) {
+        winEvents.disposeAll();
         return;
       }
-
-      this.updateValue(e.clientX);
+      if (e instanceof MouseEvent && e.clientX) {
+        this.updateValue(e.clientX);
+      }
+      if (e instanceof TouchEvent && e.touches.length) {
+        this.updateValue(e.touches[0].clientX);
+      }
       this.isActive = false;
-
       winEvents.disposeAll();
     };
+    const moveAction = (e: MouseEvent | TouchEvent) => {
+      if (!this.isActive) {
+        return;
+      }
+      this.updateValue(e instanceof MouseEvent ? e.clientX : e.touches[0].clientX);
```

Comment:
if we are only deal with first touch, maybe check if to see if there's more than 1 touch? So we can bail. Imagine the touch than drag this slider could be 2nd one.
```ts
if (isMouseEvent) {
  ....
} else {
  const touches = (e as TouchEvent).touches;
  // more than 1 finger, not sure what to do?
  if (touches.length > 1) {
    return;
  }
}
```

Reply:
@bigopon https://github.com/aurelia/ux/pull/232/commits/6af03b54495ae307905c35b69248d9e6c91f2086